src/library.js

```diff
@@ -3125,11 +3125,11 @@ addToLibrary({
 #endif
       onload(new Uint8Array(arrayBuffer));
       if (dep) removeRunDependency(dep);
-    }, (event) => {
+    }, (err) => {
       if (onerror) {
-        onerror();
+        onerror(err);
       } else {
-        throw `Loading data file "${url}" failed.`;
+        throw err;
```

Comment:
Perhaps it makes sense to keep the text here? The context could be nice. I'm not sure if there's a way to append more text to an existing Error object, though...

Reply:
Look at the OP/commit message, readAsync already has the accurate error information, this does not add any additional information, quite an opposite, it has less information.

Reply:
Interesting. So the text "Loading data file .. failed" was swallowed somewhere, I guess? Anyhow, sounds good that this improves things.